libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -15,6 +15,8 @@
 
 extern AP_HAL::HAL& hal;
 
+Compass Compass::frontend;
+
```

Comment:
If I can give my input here, I like more the way Lucas is proposing. As Sid said, the way it is presented in this PR isn't wrong, but for me it seems weird, doesn't exactly seem a singleton. This is just a personal code style preference, I don't have any technical reason for choosing one or the other.